include/bluetooth/audio/mics.h

```diff
@@ -0,0 +1,366 @@
+ *
+ * Used for to represent a Microphone Input Control Service, for either
+ * a client or a server instance. The instance pointers either represent local
+ * server instances, or remote service instances.
+ */
+struct bt_mics {
+	/** Number of Audio Input Control Service instances */
+	uint8_t aics_cnt;
+	/** Array of pointers to Audio Input Control Service instances */
+	struct bt_aics **aics;
+};
+
+/**
+ * @brief Initialize the Microphone Input Control Service
+ *
+ * This will enable the service and make it discoverable by clients.
+ * This can only be done as the server.
+ *
+ * @param param  Pointer to an initialization structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_mics_register(struct bt_mics_register_param *param);
+
+/**
+ * @brief Get Microphone Input Control Service pointer
+ *
+ * Returns a pointer to a struct that contains information about the
+ * Microphone Input Control Service instance, such as pointers to the
+ * Audio Input Control Service instances.
+ *
+ * @param conn          Connection to peer device, or NULL to get server value.
+ * @param[out] service  Pointer to store the result in.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_mics_get(struct bt_conn *conn, struct bt_mics *service);
+
+/**
+ * @brief Callback function for @ref bt_mics_discover.
+ *
+ * This callback is only used for the client.
+ *
+ * @param conn         The connection that was used to discover
+ *                     Microphone Input Control Service.
+ * @param err          Error value. 0 on success, GATT error or errno on fail.
+ * @param aics_count   Number of Audio Input Control Service instances on
+ *                     peer device.
+ */
+typedef void (*bt_mics_discover_cb)(struct bt_conn *conn, int err,
+				    uint8_t aics_count);
+
+/**
+ * @brief Callback function for Microphone Input Control Service mute.
+ *
+ * Called when the value is read,
+ * or if the value is changed by either the server or client.
+ *
+ * @param conn     Connection to peer device, or NULL if local server read.
+ * @param err      Error value. 0 on success, GATT error or errno on fail.
+ *                 For notifications, this will always be 0.
+ * @param mute     The mute setting of the Microphone Input Control Service.
+ */
+typedef void (*bt_mics_mute_read_cb)(struct bt_conn *conn, int err,
+				     uint8_t mute);
+
+/**
+ * @brief Callback function for Microphone Input Control Service mute/unmute.
+ *
+ * @param conn      Connection to peer device, or NULL if local server read.
+ * @param err       Error value. 0 on success, GATT error or errno on fail.
+ * @param req_val   The requested mute value.
+ */
+typedef void (*bt_mics_mute_write_cb)(struct bt_conn *conn, int err,
+				      uint8_t req_val);
+
+struct bt_mics_cb {
+	bt_mics_mute_read_cb            mute;
+
+#if defined(CONFIG_BT_MICS_CLIENT)
+	bt_mics_discover_cb             discover;
+	bt_mics_mute_write_cb           mute_write;
```

Comment:
I see that the AICS has separate callbacks for the mute and unmute commands - here there is a shared one.
That seems a sligh inconsistency.

Reply:
Agreed; the services are significantly different, but that doesn't mean the APIs should be :) 

Changed to be similar to AICS

Reply:
What was changed?

Reply:
Modifed the callback to not contain the mute value, but added another `unmute_write` callback

Reply:
As my mother used to say: "It is the eyes that go blind first" ...
(I really would like Github to have a feature to see changes between pushes in PRs.)